apps/documentation/src/components/TableOfContents/TableOfContents.tsx

```diff
@@ -1,115 +1,115 @@
-/*
- * Component inspired by: https://github.com/mantinedev/mantine/tree/master/apps/mantine.dev/src/components/TableOfContents
- */
-import type { JSX } from 'react'
-import { useEffect, useRef, useState } from 'react'
-import { useRouter, Link } from 'tuono'
-import { IconList } from '@tabler/icons-react'
-import { Box, rem, ScrollArea, Text } from '@mantine/core'
+import type { JSX, MouseEvent } from 'react'
+import { useRef, useState, useEffect } from 'react'
+import { useRouter } from 'tuono'
+import { Box, Text } from '@mantine/core'
 
 import { getHeadings, type Heading } from './getHeadings'
 import classes from './TableOfContents.module.css'
 
-function getActiveElement(rects: Array<DOMRect>): number {
-  if (rects.length === 0) {
-    return -1
-  }
+export function TableOfContents(): JSX.Element | null {
+  const [active, setActive] = useState<number | null>(null)
```

Comment:
This should be named `activeHedingIndex`. 
`active` is usually used for `boolean` variables